### DISCUSSION TITLE
Add bloom_filter_agg and might_contain SparkSql function

### DISCUSSION THREAD

**Discussion by jinchengchenghh** [facebookincubator/velox#3342] [Ideas]
As https://github.com/apache/spark/pull/35789 describes, it will have performance enhancement.
So I will try to implement these functions in velox.
Spark use xxhash64(col) whose return type hashcode is int64_t while velox BloomFilter only support uint64_t hashcode as hashInput https://github.com/facebookincubator/velox/blob/main/velox/common/base/BloomFilter.h#L65

- First step:

hashInput false, hashcode = folly::hasher(xxhash64(col))

- Final step:

hashInput true, hashcode = xxhash64(col)
Need to implement a stronger BloomFilter as Spark https://github.com/apache/spark/blob/master/common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilterImpl.java

The new BloomFilter should adapt hash function number according to estimatedNumItems(rowCount) and accept more column types, uint64_t and int64_t at least.

**Comment by mbasmanova** [comment]
@jinchengchenghh FYI, there is an effort to document Spark functions: https://github.com/facebookincubator/velox/pull/3890

It would be nice to add these new functions to the documentation.
